core/git_command.py

```diff
@@ -90,6 +90,8 @@ def communicate(self, stdin):
         stderr_thread.start()
 
         self.process.wait()
+        stdout_thread.join()
+        stderr_thread.join()
```

Comment:
I think we can also provide [a timeout argument][1] here (asme as in the cpython source linked), to avoid ongoing communication from hanging indefinitely. What do you think of using the process method for this calculation? E.g:

```python
savvy_settings = sublime.load_settings("GitSavvy.sublime-settings")
max_run_time = savvy_settings.get("live_panel_output_timeout", 10000)  # 10s timeout default

stdout_thread.join(self.process._remaining_time(max_run_time))
stderr_thread.join(self.process._remaining_time(max_run_time))
```

What do you think?

[1]: https://docs.python.org/3/library/threading.html#threading.Thread.join

Reply:
Agree